tests/run-pass/panic/catch_panic.rs

```diff
@@ -69,10 +69,41 @@ fn main() {
 
     // libcore panics from shims.
     #[allow(deprecated, invalid_value)]
-    test(
-        Some("Attempted to instantiate uninhabited type !"),
-        |_old_val| unsafe { std::mem::uninitialized::<!>() },
-    );
+    {
+        test(
+            Some("attempted to instantiate uninhabited type !"),
+            |_old_val| unsafe { std::mem::uninitialized::<!>() },
```

Comment:
Actually my plan was to make Miri just abort (with a good error) instead of panic. Abort is arguably better (the caller might not be panic-safe), but in real Rust we cannot abort with a good message and backtrace, which is why we panic instead. Miri has no such limitations, so I think abort is better.

But we can leave that to a future PR.